graph/src/components/store/mod.rs

```diff
@@ -104,13 +128,18 @@ pub struct EntityKey {
 
     /// ID of the individual entity.
     pub entity_id: Word,
+
+    pub causality_region: CausalityRegion,
```

Comment:
worth adding a small note on why this needs to be here? I think it's well explained on the main CR comment but might help anyone looking at this code. Possibly along the lines of "the CR this entity belong too, only DS with the same CR should be able to interact with it"

Reply:
Added a comment.